File: DSA/GEHNA_KHANNA_590013140/day32question1.c
#include <stdio.h>
int main() {
    int a, b;
    printf("Enter value of a: ");
    scanf("%d", &a);
    printf("Enter value of b: ");
    scanf("%d", &b);
    printf("Before swap: a = %d, b = %d\n", a, b);
    a = a ^ b;
    b = a ^ b;
    a = a ^ b;
    printf("After swap: a = %d, b = %d\n", a, b);
    return 0;
}

File: DSA/SakshamKashyap_590014183/day32/day32_Question_1.py
def swap_xor(a, b):
    print(f"Before Swap: a = {a}, b = {b}")
    a = a ^ b
    b = a ^ b
    a = a ^ b
    print(f"After Swap: a = {a}, b = {b}")
    return a, b

swap_xor(4, 7)
swap_xor(1, 2)
